llvm-external-projects/iree-dialects/include/iree-dialects/Transforms/TransformMatchers.h

```diff
@@ -69,14 +70,31 @@ struct NumEqualsTo : public SingleValuePredicateParam<size_t> {
   using Base::Base;
 };
 
+/// Indicates that the number of entities must be greater than the given value.
+struct NumGreaterEqualTo : public SingleValuePredicateParam<size_t> {
+  using Base::Base;
+};
+
+/// Indicates that the number of entities must be greater than the given value.
+struct NumLowerEqualTo : public SingleValuePredicateParam<size_t> {
+  using Base::Base;
+};
+
 /// Indicates that the bit width of the elemental type must be equal to the give
 /// value.
 struct ElementTypeBitWidth : public SingleValuePredicateParam<size_t> {
   using Base::Base;
 };
 
 /// Predicate tag indicating that the affine map is a permutation.
-struct IsPermutation {};
+struct IsPermutation {
+  IsPermutation(){};
+  IsPermutation &setProjectedPermutation(bool val) {
```

Comment:
I'd much rather add a new class `IsProjectedPermutation` instead of having a flag would require mixing two unrelated fluent sequences.